### PR TITLE
Name sanitation

### PR DESCRIPTION
**Pull request by kozlov721**
## Purpose

Some tools break when the network name / path contains illegal characters.
## Specification

Sanitizes the names / paths before the conversion starts.

## Dependencies & Potential Impact

None / not applicable

## Deployment Plan

None / not applicable

## Testing & Validation

None / not applicable

### CODE REVIEW

File: modelconverter/utils/general.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+def sanitize_net_name(name: str) -> str:
+    """Sanitize net name since only alphanumeric chars, hyphens and
+    underscores are allowed."""
+    if re.search(r"[^a-zA-Z0-9_-]", name):
+        sanitized = re.sub(r"[^a-zA-Z0-9_-]", "_", name)
+        logger.warning(
+            f"Illegal characters detected in: '{name}'. Replacing with '_'. New name: '{sanitized}'"
+        )
+        return sanitized
+    return name
```
Comment (ptoupas) [inline modelconverter/utils/general.py]:
```suggestion
import re
from pathlib import Path

from loguru import logger


def _normalize_underscores(s: str) -> str:
    return re.sub(r"_+", "_", s).strip("_")


def sanitize_net_name(name: str, with_suffix: bool = False) -> str:
    """Sanitize net name or path. If input is a path, only sanitize the basename. If input is a name, sanitize the whole string. Collapse multiple underscores.

    @type name: str
    @param name: The name or path to sanitize.
    @type with_suffix: bool
    @param with_suffix: If True, the suffix (file extension) is preserved and not sanitized.
    """
    p = Path(name)
    base, stem, suffix = p.name, p.stem, p.suffix

    if len(p.parts) > 1:
        if with_suffix and suffix:
            sanitized_stem = re.sub(r"[^a-zA-Z0-9_-]", "_", stem)
            sanitized_stem = _normalize_underscores(sanitized_stem)
            sanitized_base = sanitized_stem + suffix
        else:
            sanitized_base = re.sub(r"[^a-zA-Z0-9_-]", "_", base)
            sanitized_base = _normalize_underscores(sanitized_base)
        if sanitized_base != p.name:
            logger.warning(
                f"Illegal characters detected in: '{p.name}'. Replacing with '_'. New name: '{sanitized_base}'"
            )
        return (
            str(p.parent / sanitized_base)
            if str(p.parent) != "."
            else sanitized_base
        )

    if with_suffix and suffix:
        sanitized_stem = re.sub(r"[^a-zA-Z0-9_-]", "_", stem)
        sanitized_stem = _normalize_underscores(sanitized_stem)
        sanitized = sanitized_stem + suffix
    else:
        sanitized = re.sub(r"[^a-zA-Z0-9_-]", "_", name)
        sanitized = _normalize_underscores(sanitized)
    if sanitized != name:
        logger.warning(
            f"Illegal characters detected in: '{name}'. Replacing with '_'. New name: '{sanitized}'"
        )
    return sanitized

```

The changes on the `sanitize_net_name` above deal with some edge cases, and also add the `with_suffix` attribute to sanitize ignoring the suffix.

File: modelconverter/packages/base_exporter.py [hunk 2]
```diff
@@ -39,7 +44,8 @@ def __init__(
         self.disable_onnx_simplification = config.disable_onnx_simplification
         self.disable_onnx_optimization = config.disable_onnx_optimization
 
-        self.model_name = self.input_model.stem
+        self.model_name = sanitize_net_name(input_model.stem)
+        self.original_model_name = input_model.name
```
Comment (ptoupas) [inline modelconverter/packages/base_exporter.py]:
```suggestion
        self.original_model_name = sanitize_net_name(
            input_model.name, with_suffix=True
        )
```

That is so the final file moved from the intermediate_outputs has the correct name with the correct suffix (following the suggestion on the `sanitize_net_name` method.

**Review by ptoupas — CHANGES_REQUESTED** [review]
I've added some suggestions fixing some edge cases with the `output_dir` path and the final name of the file being copied from the `intermediate_outputs` folder.
Other than that it looks good to me, tested it and seems to be working as expected (after the suggested changes)

**Comment by kozlov721** [comment]
> I've added some suggestions fixing some edge cases

Thanks, I accepted the changes

**Review by klemen1999 — APPROVED** [review]

**Review by ptoupas — APPROVED** [review]
LGTM, as long as the failing RVC2 tests are fine to ignore.
